src/sd.c

```diff
@@ -284,39 +272,63 @@ uint8_t sd_erase(void)
                 continue;
             }
 
-            char file[256] = {0};
-            memcpy(file, "0:", 2);
-            memcpy(file + 2, pc_fn, (strlen(pc_fn) < 256 - 2) ? strlen(pc_fn) : 256 - 2);
-
-            FIL file_object;
-            file[0] = LUN_ID_SD_MMC_0_MEM + '0';
-            res = f_open(&file_object, (char const *)file, FA_OPEN_EXISTING | FA_WRITE);
-            if (res != FR_OK) {
-                commander_fill_report("sd_erase", FLAG_ERR_SD_OPEN, ERROR);
-                failed = 1;
-                break;
-            }
-
-            DWORD f_ps, fsize = file_object.fsize;
-            for (f_ps = 0; f_ps < fsize; f_ps++) {
-                f_putc(0xAC, &file_object); // overwrite data
+            char f_object[1024];
+            snprintf(f_object, sizeof(f_object), "%s/%s", path, pc_fn);
+
+            if (fno.fattrib & AM_DIR) { // is a directory
+                failed += delete_files(f_object);
```

Comment:
Not sure if `f_unlink(".")` is problematic. Just browsed some code and found that most devs try to skip `'.'` (https://github.com/sonicyang/uRock/blob/0a8cce2925d860bf71a3412b1b9ffc17dd09a964/Middlewares/Third_Party/FatFs/doc/img/app2.c#L26).

But if we say we delete a folder (or the whole SDCard), it would require to **NOT** skip `.<customfile>`